.github/workflows/tests.yaml

```diff
@@ -0,0 +1,59 @@
+name: Tests
+
+on:
+  workflow_run:
+    workflows: ['Code Checks']
+    types:
+      - completed
+
+jobs:
+  setup:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Set up Node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version: '20'
+
+      - name: Cache node modules
+        uses: actions/cache@v2
+        with:
+          path: ~/.npm
+          key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-node-
+
+      - name: Install dependencies
+        run: npm install
+
+  unit-tests:
+    needs: [setup]
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Run Jest tests
+        uses: willcaul/jest-github-action@v4
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          test-command: 'npm run test:jest'
+
+  e2e-tests:
```

Comment:
check this: https://github.com/avantifellows/quiz-frontend/blob/2b6643030bd3edd274266e2193cdce9535a2a636/.github/workflows/ci.yml#L35

try multiple browsers

Reply:
Adding those also.